### PR TITLE
introduces SortedMultimap interface

### PR DESCRIPTION
**Pull request by ruslansennov**
see #1694 #1886

### CODE REVIEW

File: javaslang/src/main/java/javaslang/collection/SortedMap.java [hunk 1]
```diff
@@ -71,8 +71,10 @@
     /**
      * Returns the underlying key-comparator which defines the order of the elements contained in this map.
      *
+     * @deprecated Use {@link SortedMap#comparator()} instead
      * @return This map's key-comparator.
      */
+    @Deprecated
```
Comment (ruslansennov) [inline javaslang/src/main/java/javaslang/collection/SortedMap.java]: @danieldietrich let's leave existing extension. 
Here we just mark `keyComparator()` with `@Deprecated` annotation

File: javaslang/src/main/java/javaslang/collection/TreeMultimap.java [hunk 2]
```diff
@@ -27,7 +28,7 @@
  * @author Ruslan Sennov
  * @since 2.1.0
  */
-public final class TreeMultimap<K, V> extends AbstractMultimap<K, V, TreeMultimap<K, V>> implements Serializable {
+public final class TreeMultimap<K, V> extends AbstractMultimap<K, V, TreeMultimap<K, V>> implements Serializable, SortedMultimap<K, V> {
```
Comment (ruslansennov) [inline javaslang/src/main/java/javaslang/collection/TreeMultimap.java]: no public API changed

File: javaslang/src/main/java/javaslang/collection/AbstractMultimap.java [hunk 3]
```diff
@@ -543,7 +548,10 @@ public String toString() {
 
     @Override
     public java.util.Map<K, Collection<V>> toJavaMap() {
-        final java.util.Map<K, Collection<V>> javaMap = new java.util.HashMap<>();
+        return toJavaMap(new java.util.HashMap<>());
+    }
+
+    protected <JM extends java.util.Map<K, Collection<V>>> JM toJavaMap(JM javaMap) {
```
Comment (ruslansennov) [inline javaslang/src/main/java/javaslang/collection/AbstractMultimap.java]: also used by `TreeMultimap`

File: javaslang/src/main/java/javaslang/collection/SortedMap.java (hunk 1 above)
Comment (danieldietrich) [inline javaslang/src/main/java/javaslang/collection/SortedMap.java]: Do you think it is better to carry our deprecated methods into the next major version? We have several breaking API changes. Removing all unnecessary things makes Javaslang simpler. However, we should provide a migration plan to our users.

File: javaslang/src/main/java/javaslang/collection/SortedMultimap.java [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+    @Override
+    SortedMultimap<K, V> takeRight(int n);
+
+    @Override
+    SortedMultimap<K, V> takeUntil(Predicate<? super Tuple2<K, V>> predicate);
+
+    @Override
+    SortedMultimap<K, V> takeWhile(Predicate<? super Tuple2<K, V>> predicate);
+
+}
```
Comment (danieldietrich) [inline javaslang/src/main/java/javaslang/collection/SortedMultimap.java]: Nice one!

**Pull request merged**